### PR TITLE
[SD-2472] - Error handling in fetching empty package

### PR DESCRIPTION
**Pull request by akintolga**
This wouldn't happen normally, there are few examples of packages coming from Reuters having Empty ResidRef fields in the groups in ingest collection. Handles this problem gracefully.

### CODE REVIEW

File: server/apps/duplication/archive_fetch.py [hunk 1]
```diff
@@ -111,19 +111,19 @@ def fetch(self, docs, id=None, **kwargs):
...
                 for group in dest_doc.get('groups', [])
-                for ref in group.get('refs', []) if 'residRef' in ref]
+                for ref in group.get('refs', []) if ref.get('residRef')]
 
         if refs:
             new_ref_guids = self.fetch(refs, id=None, notify=False)
             count = 0
             for ref in [ref for group in dest_doc.get('groups', [])
-                        for ref in group.get('refs', []) if 'residRef' in ref]:
+                        for ref in group.get('refs', []) if ref.get('residRef')]:
```
Comment (sivakuna-aap) [inline server/apps/duplication/archive_fetch.py]: Constant for `residRef`

**Comment by sivakuna-aap** [comment]
Can we have test for this scenario?

**Comment by mugurrus** [comment]
:+1:

**Comment by akintolga** [comment]
This PR needs https://github.com/superdesk/superdesk-core/pull/21

**Pull request merged**